source/common/common/hash.h

```diff
@@ -26,6 +27,42 @@ class HashUtil {
     return XXH64(input.data(), input.size(), seed);
   }
 
+  /**
+   * Return 64-bit hash from deterministically serializing a value in
+   * an endian-independent way and hashing it with the xxHash algorithm.
```

Comment:
It's for any context where consistent hashing might be important. It's *highly unlikely* in most contexts, but, to pick a plausible but daft example, one might have a filesystem-mounted cache using a remote filesystem and envoy instances running on different-endian machines sharing that filesystem - in this case you'd want them to use the same keys, and those keys are currently generated by hashing a proto.

Definitely makes the tests easier too!

Reply:
Alternative justification for caring - the *previous* implementation of proto hashing got endian-independent hashes because it was hashing stringifications of numbers. So "in order to not lose a property than the prior implementation had" could be a good answer to the question too.